modules/files/files.go

```diff
@@ -74,6 +74,14 @@ func CopyTerragruntFolderToTemp(folderPath string, tempFolderPrefix string) (str
 // CopyFolderToTemp creates a copy of the given folder and all its filtered contents in a temp folder
 // with a unique name and the given prefix.
 func CopyFolderToTemp(folderPath string, tempFolderPrefix string, filter func(path string) bool) (string, error) {
+	exists, err := FileExistsE(folderPath)
+	if err != nil {
+		return "", err
+	}
+	if !exists {
+		return "", os.ErrNotExist
```

Comment:
NIT: Could you create a new error that includes the path of the non-existent folder in the error message? That'll help quite a bit with debugging.

Reply:
fixed this to include the folderPath. I wonder if I should also use that same error message [here](https://github.com/gruntwork-io/terratest/pull/743/files/5ade4a1b1766a64344ef183e05f64fd59147f71b#diff-822ca59b3639b5fce8df6421bcef822e23c19fcab3e497cfc423a9dfbb1e896a)

Reply:
Yes please! In fact, we typically define a new type that implements the `error` interface (has an `Error()` method), so that (a) you can check if the error you got back was of a specific type, rather than doing string parsing and (b) you can re-use that error type in more than one place.

Reply:
got it, I created `DirNotFoundError`. Let me know if that's what you were thinking!